### ISSUE TITLE
add [proxy_ssl_name]

### ISSUE THREAD

**Issue by f4nff** [caddyserver/caddy#2483]
```
upstream serlist {
server 104.16.70.234:443;
server 104.16.72.234:443;
server 104.16.69.234:443;
server 104.16.71.234:443;
server 104.16.68.234:443;
}

server {
listen 80;

location /{
proxy_pass	https://serlist;
proxy_set_header	Host “api.cloudflare.com”;
proxy_ssl_session_reuse on;
proxy_ssl_protocols	TLSv1.2;
proxy_ssl_verify	off;
proxy_send_timeout	60s;
proxy_ssl_server_name	on;
proxy_ssl_name	“api.cloudflare.com”;	
proxy_redirect	off;
proxy_set_header	X-Real-IP $remote_addr;
proxy_set_header	X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_http_version	1.1;
proxy_set_header	Upgrade $http_upgrade;
proxy_set_header	Connection “upgrade”;
}
}
```

After testing, caddyserver can not achieve the configuration requirements above nginx.
If you don't set proxy_ssl_name, you will get a **403 Forbidden cloudflare-nginx**.

Caddy can customize the host, but the reverse proxy https, you also need to be able to customize the sni name, otherwise you can not achieve the above nginx configuration requirements.

**Comment by f4nff** [comment]
After my tests, **ca_certificates** did not meet their needs.

**Comment by hueyless** [comment]
header_upstream Host {host} ?

**Comment by f4nff** [comment]
**Header_upstream Host {host}** is http layer,
**Proxy_ssl_name** is the sni of tls,
It is different.

**Comment by hueyless** [comment]
I need Proxy_ssl_name too

**Comment by mholt** [comment]
Implemented for v2 in 8715a28.
